### ISSUE TITLE
linking between SoftwareSourceCode and SoftwareApplication?

### ISSUE THREAD

**Issue by cboettig** [codemeta/codemeta#198]
IMHO, schema.org/SoftwareSourceCode and schema.org/SoftwareApplication ought to have a reversible property indicating the relationship between the two.  This would allow us to nest properties which are technically part of SoftwareApplication and not part of SoftwareSourceCode (e.g. `memoryRequirements`) as part of a SoftwareApplication type object, rather than putting them directly as properties of the SoftwareSourceCode object.  

SoftwareSourceCode does have `targetApplication`, but this does not seem to refer to the application that the source-code is defining, but rather a runtime environment (such as an operatingSystem).  SoftwareApplication does not seem to have any property corresponding to identifying the application's source code. 

Thoughts on this?  Shall we open this issue over in https://github.com/schemaorg/schemaorg for discussion?

**Comment by ProgVal** [comment]
> SoftwareSourceCode does have `targetApplication`

Do you mean `targetProduct`?

> but this does not seem to refer to the application that the source-code is defining, but rather a runtime environment (such as an operatingSystem).

The runtime env is referred to by `runtimePlatform`.

`targetProduct` description is vague enough to mean either of this definitions

**Comment by cboettig** [comment]
right, I meant `targetProduct`.  Yes, I agree that `targetProduct`'s definition is hopelessly vague:

> Target Operating System / Product to which the code applies. If applies to several versions, just the product name can be used.

the OS and the Product (application) to built by the code means totally different things to most developers!  (Though perhaps the author of this meant the sense in which a hardware product is often used as a synonym for an OS -- i.e. `"targetProduct": "iphone"` vs `"targetProduct": "iOS"`).  But who knows, like you say, it is pretty vague.  In any event, it would be nice to have an unambiguous property on `SoftwareSourceCode` and a reverse-property on `softwareApplication` connecting these two.

**Comment by dgarijo** [comment]
At the Scientific Software Registry Collaboration Workshop we have also encountered this issue. Many people were confused because SoftwareApplication and SSC are kind of merged in the same object (I know they aren't but many people confuse both). 

In my work, I have used Codemeta in https://w3id.org/okn/o/sd and my own registry, and I had to add the property "hasSourceCode" which links a Software to its SourceCode (inverse of targetProduct, which links a source code to its SoftwareApplication). If something similar is added in codemeta (or suggested in schema.org), then the problem should be solved. 

+1 to add such a property. I will suggest it in schema.org too

**Comment by mbjones** [comment]
Sounds like a great addition to me.

**Comment by dgarijo** [comment]
Then since I don't know how much time will it take for schema.org to adopt it (more than 900 issues are still open), I suggest this property to be incorporated in codemeta first.

**Comment by mbjones** [comment]
Awesome.  A pull request would be welcome with the new term definition for the new property (hopefully aligned with an existing well-defined vocabulary if it exists) and additions to the codemeta context file.

**Comment by dgarijo** [comment]
Will propose something this week

**Comment by dgarijo** [comment]
Hello, sorry for the delay. I managed to submit the PR now including an example. I think the example will guide someone from the Software registry side.  I hope it helps!
Please let me know if you  have concerns or edits. Happy to discuss

**Comment by moranegg** [comment]
On the SCIWG we have the CodeMeta task force, with which we are working on the adoption of the codemeta terms into schema.org.
I have just added `hasSourceCode` to the list of terms.

**Comment by progval** [comment]
Resolved by #300.
